openedx/core/djangoapps/lang_pref/middleware.py

```diff
@@ -53,10 +52,6 @@ def process_request(self, request):
                 accept_header = cookie_lang
             request.META[LANGUAGE_HEADER] = accept_header
 
-            # Allow the new cookie setting to update the language in the user's session
```

Comment:
Again, can you explain why this is not needed?

Reply:
I'm still not entirely clear on this transition, but if you have the test coverage you want and all is working, I guess this would be ok.

Reply:
Prior to Django 3.2, Django was using sessions and cookies for deciding on the language of a specific page, but in Django 3.2 Django decided to use cookies only, so we made that transition during Django 3.2 upgrade (https://github.com/openedx/edx-platform/pull/28796). But we made this PR to support both Django 2.2 and 3.2 as the platform was on 2.2 back then so we didn't remove sessions completely from here and forgot to do cleanup after the upgrade. So this is basically cleaning up the session's code as this has no effect now.

https://docs.djangoproject.com/en/3.2/topics/i18n/translation/#how-django-discovers-language-preference